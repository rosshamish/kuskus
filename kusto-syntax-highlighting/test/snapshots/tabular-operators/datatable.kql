// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/datatableoperator

datatable(Date:datetime, Event:string, MoreData:dynamic) [
    datetime(1910-06-11), "Born", dynamic({"key1":"value1", "key2":"value2"}),
    datetime(1930-01-01), "Enters Ecole Navale", dynamic({"key1":"value3", "key2":"value4"}),
    datetime(1953-01-01), "Published first book", dynamic({"key1":"value5", "key2":"value6"}),
    datetime(1997-06-25), "Died", dynamic({"key1":"value7", "key2":"value8"}),
]
| where strlen(Event) > 4
| extend key2 = MoreData.key2
