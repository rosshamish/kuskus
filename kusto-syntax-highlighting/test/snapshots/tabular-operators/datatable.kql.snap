>// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/datatableoperator
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>
>datatable(Date:datetime, Event:string, MoreData:dynamic) [
#^^^^^^^^^ source.kusto keyword.other.query.kusto
#         ^^^^^^ source.kusto
#               ^^^^^^^^ source.kusto storage.type.kusto
#                       ^^^^^^^^ source.kusto
#                               ^^^^^^ source.kusto storage.type.kusto
#                                     ^^^^^^^^^^^ source.kusto
#                                                ^^^^^^^ source.kusto storage.type.kusto
#                                                       ^^^^ source.kusto
>    datetime(1910-06-11), "Born", dynamic({"key1":"value1", "key2":"value2"}),
#^^^^ source.kusto
#    ^^^^^^^^ source.kusto storage.type.kusto
#            ^ source.kusto
#             ^^^^ source.kusto constant.numeric.kusto
#                 ^ source.kusto
#                  ^^ source.kusto constant.numeric.kusto
#                    ^ source.kusto
#                     ^^ source.kusto constant.numeric.kusto
#                       ^^^ source.kusto
#                          ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                           ^^^^ source.kusto string.quoted.double.kusto
#                               ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                                ^^ source.kusto
#                                  ^^^^^^^ source.kusto storage.type.kusto
#                                         ^ source.kusto
#                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto string.other.kusto
#                                                                            ^^^ source.kusto
>    datetime(1930-01-01), "Enters Ecole Navale", dynamic({"key1":"value3", "key2":"value4"}),
#^^^^ source.kusto
#    ^^^^^^^^ source.kusto storage.type.kusto
#            ^ source.kusto
#             ^^^^ source.kusto constant.numeric.kusto
#                 ^ source.kusto
#                  ^^ source.kusto constant.numeric.kusto
#                    ^ source.kusto
#                     ^^ source.kusto constant.numeric.kusto
#                       ^^^ source.kusto
#                          ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                           ^^^^^^^^^^^^^^^^^^^ source.kusto string.quoted.double.kusto
#                                              ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                                               ^^ source.kusto
#                                                 ^^^^^^^ source.kusto storage.type.kusto
#                                                        ^ source.kusto
#                                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto string.other.kusto
#                                                                                           ^^^ source.kusto
>    datetime(1953-01-01), "Published first book", dynamic({"key1":"value5", "key2":"value6"}),
#^^^^ source.kusto
#    ^^^^^^^^ source.kusto storage.type.kusto
#            ^ source.kusto
#             ^^^^ source.kusto constant.numeric.kusto
#                 ^ source.kusto
#                  ^^ source.kusto constant.numeric.kusto
#                    ^ source.kusto
#                     ^^ source.kusto constant.numeric.kusto
#                       ^^^ source.kusto
#                          ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                           ^^^^^^^^^^^^^^^^^^^^ source.kusto string.quoted.double.kusto
#                                               ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                                                ^^ source.kusto
#                                                  ^^^^^^^ source.kusto storage.type.kusto
#                                                         ^ source.kusto
#                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto string.other.kusto
#                                                                                            ^^^ source.kusto
>    datetime(1997-06-25), "Died", dynamic({"key1":"value7", "key2":"value8"}),
#^^^^ source.kusto
#    ^^^^^^^^ source.kusto storage.type.kusto
#            ^ source.kusto
#             ^^^^ source.kusto constant.numeric.kusto
#                 ^ source.kusto
#                  ^^ source.kusto constant.numeric.kusto
#                    ^ source.kusto
#                     ^^ source.kusto constant.numeric.kusto
#                       ^^^ source.kusto
#                          ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                           ^^^^ source.kusto string.quoted.double.kusto
#                               ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                                ^^ source.kusto
#                                  ^^^^^^^ source.kusto storage.type.kusto
#                                         ^ source.kusto
#                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto string.other.kusto
#                                                                            ^^^ source.kusto
>]
#^^ source.kusto
>| where strlen(Event) > 4
#^^ source.kusto
#  ^^^^^ source.kusto keyword.other.query.kusto
#       ^ source.kusto
#        ^^^^^^ source.kusto support.function.kusto
#              ^^^^^^^^^^ source.kusto
#                        ^ source.kusto constant.numeric.kusto
>| extend key2 = MoreData.key2
#^^ source.kusto
#  ^^^^^^ source.kusto keyword.other.query.kusto
#        ^^^^^^^^^^^^^^^^^^^^^^ source.kusto
>